include/gabia/secure_socket.hpp

```diff
@@ -213,7 +228,12 @@ class secure_socket {
             CORO_REENTER(*this) {
                 while (transferred < sizeof(payload_length) + payload_length +
                                          crypto::aead_auth_tag_size) {
-                    raw_sequence = socket.raw_buffer.prepare(1024);
+                    raw_sequence = prepare_or_error(socket.raw_buffer, 1024, error);
+                    if (error) {
+                        // TODO: Should the socket be closed? It's going to be most likely in an unusable state.
+                        socket.get_io_service().post(beast::bind_handler(handler, error, 0));
```

Comment:
Once an error (other than `eof`) is detected, Beast never attempts to read or write to the stream. It returns as quickly as possible.